tests/testsuite/install.rs

```diff
@@ -2730,3 +2730,20 @@ fn uninstall_running_binary() {
 
 "#]]).run();
 }
+
+#[cargo_test]
+fn dry_run() {
+    pkg("foo", "0.0.1");
+
+    cargo_process("-Z unstable-options install --dry-run foo")
+        .masquerade_as_nightly_cargo(&["install::dry-run"])
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[DOWNLOADING] crates ...
+[DOWNLOADED] foo v0.0.1 (registry `dummy-registry`)
+[INSTALLING] foo v0.0.1
```

Comment:
We need an "aborting install" warning message at the end

Reply:
According to you, is it still needed to have this message ? Since the last message is now `[INSTALLED] package ...` i think that it would be misleading to have an ''aborting install" message.

Reply:
Without it, we've told the user we've installed.  If there is a concern with other messages, we can consider tweaking them

Reply:
Changed in 63a4955c7ed614df8f6f454b66c5fe9488c215e7.
I've decided to display an `[ABORTING] install` message instead of `[INSTALLED] package` to avoid confusion , are you okay with that ?

Reply:
1. I'd like the style of dry-run between commands to be consistent
2. I'd a command's output to be consistent whether dry-run or not

What are your thoughts on doing something like #13367 and switching the messages to `Installing` / `Replacing` and printing them before we act on them?

Reply:
So if i understood correctly, you want to replace the `Installed` message by an `Installing` message, before we act on them, in both dry-run an normal mode, is it this ?
Should we still keep the `[ABORTING]` message then ?

Reply:
Its one way of resolving this problem that I'm hopeful will work out.  If we go this route, we should probably break this out into its own PR so we can evaluate it on its own.

> Should we still keep the [ABORTING] message then ?

We should just do the `[WARNING]` like other dry-runs

Reply:
Ok so for now i just removed the `[ABORTING]` message and displayed a `[WARNING]` message like for `cargo add --dry-run`